MAINTAINERS.adoc

```diff
@@ -0,0 +1,129 @@
+.. `git status` should show that you're on the `main` branch.
+.. Use `git pull` to sync up with any upstream changes.  `git status` should show that your local "main" is exactly sync'd up with the upstream "main" and that you have no uncommitted files or local changes to committed files.
+.. `git clean -nxd` should be clean or close to it.  It should show nothing that should be checked into git and probably no stale build artifacts.
+.. It's a good idea to check that CI completed successfully on GitHub for the commit that you've got checked out.  This will ensure that we're publishing a release that builds from scratch, passes all tests on all platforms, and passes our style and lint checks.
+... Use `git log -1` to see what commit you're on.
+... Look up that commit on GitHub.  It will have a URL like `https://github.com/oxidecomputer/dropshot/commit/COMMIT_SHA` (with an actual `COMMIT_SHA`).  CI succeeded if there's a green check at the left of the commit message.  You can click the green check to be sure you're looking at the right thing.  It should say "All checks have passed".
+. Figure out the tag used for the most recent release.  The easiest is probably to view https://github.com/oxidecomputer/dropshot/releases[the recent releases].   We'll call this `PREV_RELEASE_TAG` in the next step.
+. Figure out what kind of update this is relative to the last one released.  https://github.com/sunng87/cargo-release/blob/master/docs/reference.md#bump-level[See `cargo release` docs for the available levels.]  We're usually doing a "patch", "minor", or "major" release.  We'll call this "RELEASE_KIND".  (According to the docs, you can also use a specific version number that you want to release here.)
+. Do the release.  If you're daring, just run:
++
+[source,text]
+----
+$ cargo release --prev-tag-name=PREV_RELEASE_TAG -vv RELEASE_KIND
+----
++
+If you want to be extra careful, you can do this in a dry-run form.  First, check that the log output here looks right (and especially that you got the version number that you expected!):
++
+[source,text]
+----
+$ cargo release --dry-run --prev-tag-name=PREV_RELEASE_TAG -vv RELEASE_KIND
+----
++
+Now, run `cargo release` for real (i.e. no `--dry-run`), but without publishing to crates.io or pushing the git tags:
++
+[source,text]
+----
+$ cargo release --skip-push --skip-publish --prev-tag-name=PREV_RELEASE_TAG -vv minor
+----
++
+Inspect the resulting commits and the new tags from above.  There should be four commits: one to each of "dropshot" and "dropshot_endpoint" that sets the version to the one you specified, and one that sets the version to the _next_ version.  For example, if the current version is 0.5.1 and you're publishing 0.5.2, the repo will start at 0.5.2-pre, you'll see one commit that sets it to 0.5.2, and then you'll see one that sets it to 0.5.3-pre.
++
+Then undo the above step:
++
+[source,text]
+----
+$ git reset --hard THE_COMMIT_YOU_STARTED_WITH
+$ git tag -d THE_NEW_TAG
+----
```

Comment:
Since these two steps cancel each other out, why bother doing them? Doesn't the dry run provide enough information about what will happen?

Reply:
If I were going to drop either the `--dry-run` step or the `--skip-push --skip-publish` step, I would skip the `--dry-run` because it doesn't provide any information over the other one.  I believe I saw cases (while working out the config) where the dry run succeeded and the `--skip-push --skip-publish` failed.

I think what you're getting at (and I kind of alluded to later without saying it) is that there's a bunch of fear in this procedure.  I think that fear stems from my initial experiments with `cargo-release` resulting in it doing things I found very surprising.  I don't remember more than that, but the lesson I internalized was that I wanted to see _exactly_ what it was doing before committing to anything.  I expect others might not be so paranoid about it, which is why I started with the "if you're daring" one-liner, but kept in enough detail for somebody to do it very carefully if they wanted.

Reply:
Ah, yeah, at first read I think I somehow missed that "if you want to go all-out, use this one liner" option. Makes sense to have a more intensive breakdown to see all the incremental changes, and then folks can rely on the shorthand once they (or the collective we) trust the process.